modules/nf-core/phispy/tests/main.nf.test

```diff
@@ -0,0 +1,69 @@
+// nf-core modules test phispy
+nextflow_process {
+
+    name "Test Process PHISPY"
+    script "../main.nf"
+    process "PHISPY"
+
+    tag "modules"
+    tag "modules_nfcore"
+    tag "phispy"
+
+    test("bacteroides_fragilis - gbff") {
+
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test' ], // meta map
+                    file( params.modules_testdata_base_path + "/genomics/prokaryotes/bacteroides_fragilis/genome/genome.gbff.gz", checkIfExists: true )
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert path(process.out.gbk[0][1]).linesGzip.toString().contains('5150703 bp') },
+                { assert snapshot(process.out.versions).match() }
+            )
+            with(process.out.log[0]) {
```

Comment:
This should be inside the `assertAll`, no?
What is it actually testing when it's outside the `assertAll`? I need to read up on that.

Reply:
Oh I thought that wasn't necessary. The example I used as a basis for `with` (https://github.com/nf-core/modules/blob/master/modules/nf-core/custom/sratoolsncbisettings/tests/main.nf.test) didn't have an assertAll.

Reply:
It's not about which tests are run, but what is considered a pass. Can the `with` part fail for example, and the test will still result in success?

Also the `assertAll` is rule 1 in the https://nf-co.re/docs/contributing/tutorials/nf-test_assertions#nf-core-guidelines-for-assertions

Reply:
Also not sure how to wrap the `with` block inside of `assertAll`, keeps raising a syntax error here.

Reply:
> It's not about which tests are run, but what is considered a pass. Can the `with` part fail for example, and the test will still result in success?
> 
> Also the `assertAll` is rule 1 in the https://nf-co.re/docs/contributing/tutorials/nf-test_assertions#nf-core-guidelines-for-assertions

In the tests I did, when the with part fails the whole test fails. Either way, managed to wrap the `with` statements inside `assertAll` (122ddb151) - The syntax errors were just curly braces and commas, the usual, sorry 😄

Reply:
I think without the `assertAll` it's fast-fail, so you only know the first assertion that fails. I think `assertAll` reports how many assertions fail in the group as well as which.

Reply:
Ah I see, good to know, thanks!